java/client/src/main/java/com/youtube/vitess/client/cursor/Row.java

```diff
@@ -0,0 +1,223 @@
+package com.youtube.vitess.client.cursor;
+
+import com.google.common.collect.Lists;
+import com.google.common.io.BaseEncoding;
+import com.google.common.primitives.UnsignedLong;
+import com.google.protobuf.ByteString;
+import com.youtube.vitess.proto.Query;
+import com.youtube.vitess.proto.Query.Field;
+import org.apache.hadoop.io.Writable;
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.sql.SQLDataException;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+
+public class Row implements Writable {
+  private Map<String, Integer> fieldMap;
+  private List<Field> fields;
+  private Query.Row rawRow;
+
+  public Row(List<Field> fields, Query.Row rawRow, Map<String, Integer> fieldMap) {
```

Comment:
FYI: In the C++ implementation I created an object that contains both List<Field> and FieldMap (since they usually go together). I called it Cursor::Schema.